### PR TITLE
Paul/info rollup

### PR DESCRIPTION
**Pull request by paullinator**
### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [ ] Yes
- [ ] No

### Dependencies

 none

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [ ] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207730407796189

### CODE REVIEW

File: src/__tests__/actions/AccountReferralActions.test.ts
Comment (samholmes) [inline src/__tests__/actions/AccountReferralActions.test.ts:1]: I don't know what promoCard v1 is, so I'm assuming it's safe to remove this.

**Review by samholmes — CHANGES_REQUESTED** [review]
Good stuff. I can see this really lowering our network activity on the react JS thread and also reducing renders.

File: src/components/scenes/TransactionListScene.tsx [hunk 1]
```diff
@@ -20,7 +19,7 @@ import { useSceneScrollHandler } from '../../state/SceneScrollState'
 import { config } from '../../theme/appConfig'
 import { useSelector } from '../../types/reactRedux'
 import { EdgeSceneProps } from '../../types/routerTypes'
-import { fetchInfo } from '../../util/network'
+import { infoServerData } from '../../util/network'
```
[the remark this answers is not in the document]
Comment (paullinator) [reply src/components/scenes/TransactionListScene.tsx:22]: it's no more out of date than the refresh interval of the info server loop in app.ts.

File: src/plugins/stake-plugins/thorchainSavers/tcSaversPlugin.ts [hunk 2]
```diff
@@ -259,26 +259,12 @@ export const makeTcSaversPlugin = async (opts: EdgeGuiPluginOptions): Promise<St
...
-    try {
-      const fetchResponseJson = await fetchResponse.json()
-      const infoServerResponse = asInfoServerResponse(fetchResponseJson)
-      updatePolicyApys(infoServerResponse)
-    } catch (err: any) {
-      const msg = `Parsing Fetch APY failed: ${err.message}`
-      console.warn(msg)
-    }
+  try {
+    const infoServerResponse = asInfoServerResponse(infoServerData.rollup?.apyValues)
```
[the remark this answers is not in the document]
Comment (paullinator) [reply src/plugins/stake-plugins/thorchainSavers/tcSaversPlugin.ts:263]: yep. and there's a catch.

File: src/actions/ExchangeInfoActions.ts [hunk 3]
```diff
@@ -54,13 +53,8 @@ export type ExchangeInfo = ReturnType<typeof asExchangeInfo>
...
   return async (dispatch, getState) => {
     try {
-      const reply = await fetchInfo(`v1/exchangeInfo/${config.appId ?? 'edge'}`)
-      if (!reply.ok) {
-        const text = await reply.text()
-        throw new Error(`updateExchangeInfo returned error ${text}`)
-      }
-      const clean = asExchangeInfo(await reply.json())
-      dispatch({ type: 'UPDATE_EXCHANGE_INFO', data: clean })
+      const data = asExchangeInfo(infoServerData.rollup?.exchangeInfo)
```
[the remark this answers is not in the document]
Comment (paullinator) [reply src/actions/ExchangeInfoActions.ts:56]: a cleaner throw is fine as we have a catch. That way we don't bother to update redux and cause a re-render

File: src/components/scenes/TransactionListScene.tsx [hunk 4]
```diff
@@ -142,8 +143,17 @@ function TransactionListComponent(props: Props) {
...
+      const curDate = new Date().toISOString()
+
+      const title = getLocaleOrDefaultString(localeStatusTitle)
+      const message = getLocaleOrDefaultString(localeStatusBody)
+
+      if (title == null || message == null) return false
+      if (appId != null && appId !== config.appId) return false
+      if (appVersions != null && !appVersions.includes(version)) return false
+      if (statusEndIsoDate != null && statusEndIsoDate < curDate) return false
+      if (statusStartIsoDate != null && statusStartIsoDate > curDate) return false
```
[the remark this answers is not in the document]
Comment (paullinator) [reply src/components/scenes/TransactionListScene.tsx:156]: correct. if they are null, then we don't apply that as a filter

File: src/__tests__/actions/AccountReferralActions.test.ts
Comment (paullinator) [reply src/__tests__/actions/AccountReferralActions.test.ts:1]: deprecated old endpoint

File: src/components/scenes/TransactionListScene.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (samholmes) [reply src/components/scenes/TransactionListScene.tsx:156]: Ah, I got confused with the diff. You are right that `appId` and `appVersion` was allowed to be null. Though looking at the `isWithinDate` logic, it looks like it didn't allow null for `statusEndIsoDate` and `statusStartIsoDate`. Am I reading that wrongly, or was this a bug before?

**Review by samholmes — APPROVED** [review]

File: src/components/scenes/TransactionListScene.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (paullinator) [reply src/components/scenes/TransactionListScene.tsx:156]: you're right. technically a previous bug. all promoCard params that are optional should be allowed to be `undefined` and then they simply don't get filtered by that param

[the remark this answers is not in the document]
Comment (paullinator) [reply src/components/scenes/TransactionListScene.tsx:156]: hmm. it looks like the dates aren't optional and will always be there anyway so the null checks are pointless

**Pull request merged**